sample/java/src/main/java/com/datadog/android/sample/traces/TracesViewModel.java

```diff
@@ -22,6 +31,62 @@ public void stopAsyncOperations() {
             asyncTask.cancel(true);
             asyncTask = null;
         }
+
+        if (request != null) {
+            request.cancel(true);
+            request = null;
+        }
+    }
+
+    public static class Request extends AsyncTask<Void, Void, Request.Result> {
+
+        static class Result {
+
+            public Response response;
+            public Exception exception;
+
+            public Result(Response response, Exception exception) {
+                this.response = response;
+                this.exception = exception;
+            }
+
+        }
+
+        public interface Callback {
+            void onResult(Result result);
+        }
+
+        private final Callback callback;
+
+        public Request(Callback callback) {
+            this.callback = callback;
+        }
+
+        private OkHttpClient okHttpClient = new OkHttpClient.Builder()
+                .addInterceptor(new TracingInterceptor())
+                .addNetworkInterceptor(new StethoInterceptor())
+                .build();
+
+        @Override
+        protected Result doInBackground(Void... urls) {
+            okhttp3.Request request = new okhttp3.Request.Builder()
+                    .get()
+                    .url("https://dd.datad0g.com/")
```

Comment:
shouldn't we use the *url* from the function argument here ?

Reply:
No the arguments are void it's named `urls` from earlier tests I'll change it.

Reply:
Maybe we should add all of these changes also in the Kotlin app to keep them aligned.

Reply:
We could, but eventually we'll have the shopist app to test everything, which will be easier than maintaining the same app twice.